daemon/container.go

```diff
@@ -1557,3 +1548,37 @@ func (c *Container) LogDriverType() string {
 	}
 	return c.hostConfig.LogConfig.Type
 }
+
+type archiveFilter struct {
+	basePath string
+	files    []string
+}
+
+func archiveGlob(basePath string) (*archiveFilter, error) {
+	all, err := filepath.Glob(basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(all) == 0 {
+		return nil, os.ErrNotExist
+	}
+
+	baseDir := filepath.Dir(basePath)
+	for baseDir != "" && !strings.HasPrefix(all[0], baseDir) {
```

Comment:
Imagine that the matcher is `/tmp/**/*`. You want to find the base directory common to the expanded paths, for instance `/tmp/foo/test`, and the matcher. Otherwise, you could end up using `/tmp/**` as a base directory, and that doesn't exist.